packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -34,60 +34,147 @@ describe('HTS Precompile Acceptance Tests', async function() {
     expect(await HTSTokenContract.balanceOf(accounts[2].wallet.address)).to.equal(0);
   });
 
   it('should be able to transfer hts tokens between accounts', async function() {
     const amount = 10;
-    const balanceBefore = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
-
-    await baseHTSContract.transferTokenTo(accounts[1].wallet.address, HTSTokenContract.address, amount);
+    const HTSTokenContract = new ethers.Contract(HTSTokenContractAddress, ERC20MockJson.abi, accounts[0].wallet);
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
 
+    const balanceBefore = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
+    await baseHTSContract.transferTokenPublic(accounts[1].wallet.address, HTSTokenContractAddress, amount);
     const balanceAfter = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
+
     expect(balanceBefore + amount).to.equal(balanceAfter);
   });
+
+  it('should be able to check allowance', async function() {
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = await baseHTSContract.allowancePublic(HTSTokenContractAddress, baseHTSContractAddress, accounts[2].wallet.address);
+    const { responseCode } = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args;
+
+    expect(responseCode).to.equal(TX_SUCCESS_CODE);
+  });
+
+  it('should be able to approve anyone to spend tokens', async function() {
+    const amount = 13;
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+
+    const txBefore = await baseHTSContract.allowancePublic(HTSTokenContractAddress, baseHTSContractAddress, accounts[2].wallet.address);
+    const beforeAmount = (await txBefore.wait()).events.filter(e => e.event === 'AllowanceValue')[0].args.amount.toNumber();
+
+    await baseHTSContract.approvePublic(HTSTokenContractAddress, accounts[2].wallet.address, amount, { gasLimit: 1_000_000 });
+
+    const txAfter = await baseHTSContract.allowancePublic(HTSTokenContractAddress, baseHTSContractAddress, accounts[2].wallet.address);
+    const afterAmount = (await txAfter.wait()).events.filter(e => e.event === 'AllowanceValue')[0].args.amount.toNumber();
+
+    expect(beforeAmount).to.equal(0);
+    expect(afterAmount).to.equal(amount);
+  });
+
+  it('should be able to execute setApprovalForAll on nft', async function() {
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = (await baseHTSContract.isApprovedForAllPublic(NftHTSTokenContractAddress, baseHTSContractAddress, accounts[1].wallet.address));
+    const { responseCode } = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args;
+
+    expect(responseCode).to.equal(TX_SUCCESS_CODE);
```

Comment:
Move this check into the next test `should be able to execute setApprovalForAllPublic` and you should be able to remove this test.
For unit tests it's good to have it broken out but for E2E acceptance you can't do anything without success and it calls cost HBAR so no need for the breakout